### DISCUSSION TITLE
Correctly using NetworkModule visualisation to visualise a network

### DISCUSSION THREAD

**Discussion by MechaKaradi** [mesa/mesa#1560] [Q&A] [answered]
Hey! I am new to Mesa and ABM modelling in general and am attempting to implement a disaster response model that runs on a city's street network as a grid. I am struggling to get the visualization of the model to run in the inbuilt visualization server. 

I have created the network graph using the `nx library, `and am able to view it correctly using the `draw_networkx` function. I have also defined a `network_portrayal ` function under the `server.py` file modeled after the `boltzmann_wealth_model_network ` visualization example in the examples repo. The python scripts don't give me any errors when run, however, the visualization server comes up blank and the javascript console shows: `TypeError: t.app is undefined content_script_bundle.js:93:1027`

The documentation isn't super clear about what could be wrong. Has anyone encountered this bug before? 
What could be the cause? 

I am attaching a link to my repository for this project in-case someone wants to take a look at my code. 

Thanks! 
[REP](https://github.com/MechaKaradi/EarthquakeABM-1211l)

**Comment by rht** [comment, marked as answer]
The error message I got is `Uncaught Error: node not found: 15012`.
`TypeError: t.app is undefined content_script_bundle.js:93:1027` this is probably one of your Chrome/Firefox extension's problem.

The error is due to street_network.data has the node indices starting from 1, 2, 3, ..., but D3.js assumes the nodes start from 0, 1, 2...
To fix the `node not found` error message, you rename node 15012 to node 0:
```python
self.G = nx.relabel_nodes(self.G, {3999: 0})
```
